doc/manual/nasl/nasl-grammar.md

```diff
@@ -0,0 +1,247 @@
+- `!~` is the negated regex matching operator.
+Note: `=~` and `!~` support the (standard extended POSIX) regex pattern. See [Regular Expression Functions](../built-in-functions/regular-expressions/index.md) for more background info on the supported regex expression support.
+
+### Assignment
+An assignment expression is syntactically similar to a binary expression but assigns the RHS expression to the LHS value. The LHS is restricted to valid assignment targets:
+```nasl
+a = 3; // assigns 3 to a
+a[1] = 3; // assigns 3 to the second element of a (an array).
+"foo" = 3; // ERROR: string literal is not a valid assignment target.
+```
+
+Available assignment operators:
+`=`: Simple assignment.
+`+=`: Adds the RHS to the LHS.
+`-=`: Subtracts the RHS from the LHS.
+`*=`: Multiplies the LHS by the RHS.
+`/=`: Divides the LHS by the RHS.
+`%=`: Computes (LHS % RHS) and assigns it to LHS.
+`<<=`: Computes `(LHS << RHS)` and assigns it to LHS.
+`>>=`: Computes `(LHS >> RHS)` and assigns it to LHS.
+`>>>=`: Computes `(LHS >>> RHS)` and assigns it to LHS.
+
+### Increment
+An increment expression is syntactically similar to a unary expression and is restricted to valid assignment targets as well:
+```nasl
+a++; // Increments foo by 1
+a[1]++; // Increments the second element of a (an array) by 1.
+"foo"++; // ERROR: string literal is not a valid assignment target.
+```
+
+Available increment operators:
+`++`: Increments operand by one.
+`--`: Decrements operand by one.
+
+Postfix operators (`x++` and `x--`) evaluate to the value of x, then modify the value. For example, in `x = 10; x++;`, the second statement evaluates to `10`.
+Prefix operators (`++x` and `--x`) modify the value, then evaluate to the new value. For example, in `x = 10; ++x;`, the second statement evaluates to `11`.
+
+## Operator precedence
+| Operators                       | Associativity |
+| ------------------------------- | ------------- |
+| ++ --                           | None          |
+| **                              | Right         |
+| ~- (unary minus)                | Left          |
+| = += -= *= /= %= <<= >>= >>>=   | Right         |
+| !                               | Left          |
+| * / %                           | Left          |
+| + -                             | Left          |
+| << >> >>>                       | Left          |
+| &                               | Left          |
+| ^                               | Left          |
+| \|                              | Left          |
+| < <= > >= == != <> =~ !~ >!< >< | None          |
+| &&                              | Left          |
+| \|\|                            | Left          |
+
+Note, in particular, that assignment has relatively high precedence, allowing expressions like
+```nasl
+if (!x = foo() ) { ... }
+```
+which will first assign the result of the `foo()` call to `x`, and then negate the value.
+
+## Loops and Control Flow
+### If
+If statements are written in the usual way as
```

Comment:
I would remove `in the usual way` as it depends on the experience of the reader if it is usual or not and it doesn't bring value into the sentence.